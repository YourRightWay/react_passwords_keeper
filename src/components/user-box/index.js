import UserBox from "./user-box"

export default UserBox;
