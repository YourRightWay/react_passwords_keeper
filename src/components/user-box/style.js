const styles = theme => ({
    root: theme.root,
    userBox: {
        paddingLeft: '15px',
        marginLeft: '15px',
        borderLeft: '2px solid #ccc',
    },
    userEmail: {

    },
});

export default styles;

