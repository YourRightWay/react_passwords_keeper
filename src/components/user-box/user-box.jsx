import React from 'react';
import PropTypes from 'prop-types';
import compose from 'recompose/compose';
import withWidth from 'material-ui/utils/withWidth';
import { withStyles } from 'material-ui/styles';
import styles from './style';

/**
 * Это вполне может быть stateless компонентом без методов жизненого цикла так как
 * он не имеет локального стостояния и методов
 */

// class UserBox extends Component {
//     render() {
//         const {classes, user} = this.props;
//         return(
//                 <div className={classes.userBox}>
//                     <h2>{user.name}</h2>
//                     <img src={user.photo} alt=""/>
//                     <p className={classes.userEmail}>{user.email}</p>
//                 </div>
//         )
//     }
// }

export const UserBox = ({ classes: { userBox, userEmail }, user: { name, photo, email } }) => (
    <div className={userBox}>
        <h2>{name}</h2>
        <img src={photo} alt=""/>
        <p className={userEmail}>{email}</p>
    </div>
);

// PropTypes ??

UserBox.propTypes = {
    classes: PropTypes.object.isRequired,
    user: PropTypes.object.isRequired,
};

export default compose(withStyles(styles), withWidth())(UserBox);
