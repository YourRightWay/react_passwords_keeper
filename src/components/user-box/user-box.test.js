import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { UserBox } from './user-box';

configure({ adapter: new Adapter() });

const mockData = {
    classes: {
        userBox: 'test',
        userEmail: 'kremlin@gmail.com'
    },
    user: {
        name: 'Vova Putin',
        photo: 'nttp://assets/img/path.jpg',
        email: 'kremlin@gmail.com'
    }
};

const component = (
    <UserBox {...mockData} />
);

describe('<UserBox />', () => {
    test('Snapshot <UserBox />', () => {
        const json = renderer.create(component).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('Should render an <div /> tag', () => {
        expect(shallow(component).type()).toEqual('div');
    });

    test('Should render an <h2 /> tag', () => {
        expect(shallow(component).find('h2').length).toBe(1);
    });

    test('Should render an <img /> tag', () => {
        expect(shallow(component).find('img').length).toBe(1);
    });

    test('Should render an <p /> tag', () => {
        expect(shallow(component).find('p').length).toBe(1);
    });
});
