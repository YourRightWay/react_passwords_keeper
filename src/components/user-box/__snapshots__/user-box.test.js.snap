// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBox /> Snapshot <UserBox /> 1`] = `
<div
  className="test"
>
  <h2>
    Vova Putin
  </h2>
  <img
    alt=""
    src="nttp://assets/img/path.jpg"
  />
  <p
    className="kremlin@gmail.com"
  >
    kremlin@gmail.com
  </p>
</div>
`;
